circuit_knitting/utils/transforms.py

```diff
@@ -52,15 +64,41 @@ def separate_circuit(
 ) -> SeparatedCircuits:
     """Separate the circuit into its disconnected components.
 
+    If ``partition_labels`` is provided, then the circuit will be separated
+    according to those labels.  If it is ``None``, then the circuit will be
+    fully separated into its disconnected components, each of which will be
+    labeled with consecutive integers starting with 0.
+
+    >>> qc = QuantumCircuit(4)
+    >>> _ = qc.x(0)
+    >>> _ = qc.cx(1, 2)
+    >>> _ = qc.h(3)
```

Comment:
Without the `_ = `, one gets extra output from the non-`None` expressions:

```
>>> from qiskit import QuantumCircuit
>>> qc = QuantumCircuit(4)
>>> qc.x(0)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f710e9921c0>
>>> qc.cx(1, 2)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f70da3873a0>
>>> qc.h(3)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f70da3872b0>
```

and doctest fails unless these are also added here.  Of course, the memory address is not predictable, so there is no way to allow such a thing to be output and still have the tests pass.